### DISCUSSION TITLE
Adding Codeberg to "deploying-your-docs" file

### DISCUSSION THREAD

**Discussion by Andre601** [mkdocs/mkdocs#3434] [General]
I'm currently considering to work on adding a section in the `deploying-your-docs.md` file about Codeberg Pages.

I believe it's worth mentioning this option as it may not fall under the `Other Providers` category (With the FTP, SSH, ... stuff) but also doesn't quite fit the `GitHub Pages` section itself.

Effectively, Codeberg Pages offers features similar to GitHub Pages, including Org-level sites (`user.codeberg.page`) and project-level sites (`user.codeberg.page/project`), custom domains support and such.
Tho, it also has differences that would need to be pointed out, namely:

- Branch name for project-level site is `pages` and repo name for org-level site is `pages`
- No support for a `docs` dir or alike on project-level sites (Only branch).
- Domain file is called `.domains` (Possible issues with MkDocs' default ignore behaviour)
  - `.domains` file can contain more than one (sub-)domain to use.
- Currently, project-level sites will not appear under your custom domain if you have an org-level site with a custom domain defined.
  - This means that if you have a org-level site with custom domain `example.com`, and a project-level site without any custom domain, the project-level site won't show under `example.com/project` (If repo is called `project`)
  - Additionally is `user.codeberg.page/project` automatically redirected to your custom domain.
  - This is a [reported issue](https://codeberg.org/Codeberg/pages-server/issues/236).
- Site is not "deployed" like GitHub pages. Instead does the pages-server pull from the respective repository (Based on DNS settings, if custom domain) directly and serve that. Tho, there is caching to reduce calls and rate limit problems, as it uses the Gitea API.
- 404 is not used... ever. Not even on org-level sites with custom domain. Trust me, I checked.

Would it be worth considering this addition?
Or maybe consider an overhaul of the page in general? Personally I'm more for that with a focus on a more generic "How to deploy" explanation with maybe some examples for popular choices (GitHub, GitLab, ReadTheDocs, ...)

**Comment by Andre601** [comment]
Bumping this as I'm split on this...

Personally, I tend more towards having the `deploying-your-page.md` file be reworked to suggest CI/CD solutions like GitHub Actions or Woodpecker-CI for deploying your docks instead of the more specific commands...
But I also don't want to make a PR that wouldn't be considered nor merged in the end and instead first want to discuss this a bit...

**Reply by oprypin** [reply]
@Andre601 sorry, I don't have anything good to say and I'm sad that I leave this hanging.

Basically I've been unable to focus my attention on anything that isn't "fun", so I haven't been able to write documentation even though it's in a dire state and should be among the top priorities. And at the same time if someone did write documentation, I'd find so many nitpicks in how I'd prefer to see it...

**Reply by Andre601** [reply]
> And at the same time if someone did write documentation, I'd find so many nitpicks in how I'd prefer to see it...

Then perhaps a good idea would be some page explaining how to contribute to the docs, with focus on how it is styled and all...

**Reply by oprypin** [reply]
The problem is I'd have to write that page  :joy:

**Reply by Andre601** [reply]
Yep...

I personally have considered working on the docs, tho, I would need guidance on what pages need a bit of clean up and what to look out for in that regard...
I mean, making a PR and then getting feedback on what to change and improve is part of the course of a PR... After all is a first PR barely any perfect, mergabla candidate, if it isn't simple typo corrections.

**Reply by pawamoy** [reply]
> I'd find so many nitpicks in how I'd prefer to see it

I think contributors would be fine with that. You can then either suggest changes, or directly push to the PR branch. I'd be happy to see that the maintainer is working with me to finalize the PR :slightly_smiling_face: And having a PR with initial contents could be a good motivator to work on the docs :stuck_out_tongue: And could fuel the list of things to say in the "How to contribute to the docs" page.

**Comment by oprypin** [comment]
I agree that this page needs to be reworked.

"Just use gh-deploy" isn't even the first option I'd recommend to GitHub users anymore - that'd be GitHub Actions instead. Although then there isn't an obvious exactly correct setup to recommend.

And if it's not GitHub, then yeah the page is not very useful. Actually I realize that I haven't even read this page before. I'm positively surprised that the "Local Files" section is relatively prominent. I'd actually give it a quick mention right at the beginning of the page that MkDocs just produces a directory that you can "deploy" in any way you like.

> Personally I'm more for that with a focus on a more generic "How to deploy" explanation with maybe some examples for popular choices (GitHub, GitLab, ReadTheDocs, ...)

Yeah for sure. Actually it's not even so far from this state at the moment, it *is* kinda generic (the sections names are: GitHub, ReadTheDocs, Other, Files)
